torchgeo/datasets/landcoverai.py

```diff
@@ -50,7 +52,227 @@ class LandCoverAI(NonGeoDataset):
+    cmap = {
+        0: (0, 0, 0, 0),
+        1: (97, 74, 74, 255),
+        2: (38, 115, 0, 255),
+        3: (0, 197, 255, 255),
+        4: (207, 207, 207, 255),
+    }
+
+    def __init__(
+        self, root: str = "data", download: bool = False, checksum: bool = False
+    ) -> None:
+        """Initialize a new LandCover.ai dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            cache: if True, cache file handle to speed up repeated sampling
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            RuntimeError: if ``download=False`` and data is not found, or checksums
+                don't match
+        """
+        self.root = root
+        self.download = download
+        self.checksum = checksum
+
+        lc_colors = np.zeros((max(self.cmap.keys()) + 1, 4))
+        lc_colors[list(self.cmap.keys())] = list(self.cmap.values())
+        lc_colors = lc_colors[:, :3] / 255
+        self._lc_cmap = ListedColormap(lc_colors)
+
+        self._verify()
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+
+        Raises:
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        if self._verify_data():
+            return
+
+        # Check if the zip file has already been downloaded
+        pathname = os.path.join(self.root, self.filename)
+        if os.path.exists(pathname):
+            self._extract()
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise RuntimeError(
+                f"Dataset not found in `root={self.root}` and `download=False`, "
+                "either specify a different `root` directory or use `download=True` "
+                "to automatically download the dataset."
+            )
+
+        # Download the dataset
+        self._download()
+        self._extract()
+
+    @abc.abstractmethod
+    def _verify_data(self) -> bool:
+        """Verify if the images and masks are present."""
+
+    def _download(self) -> None:
+        """Download the dataset."""
+        download_url(self.url, self.root, md5=self.md5 if self.checksum else None)
+
+    def _extract(self) -> None:
+        """Extract the dataset."""
+        extract_archive(os.path.join(self.root, self.filename))
+
+    def plot(
+        self,
+        sample: Dict[str, Tensor],
+        show_titles: bool = True,
+        suptitle: Optional[str] = None,
+    ) -> plt.Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample returned by :meth:`__getitem__`
```

Comment:
RtD tests are failing because this method isn't defined. Can you add an abstract method that the subclasses are responsible for overriding? COWC has an example of this.

Reply:
Bleh. Hmm... Here are some things you can try, in order of preference.

1. Make the input type `Union[BoundingBox, int]`
2. Make the input type `Any`
3. Use `typing.overload`
4. Give up on abc's and convert ``:meth:`__getitem__` `` to ``` ``__getitem__`` ```

Reply:
It has to be Any. Otherwise, mypy returns an error related to inheritance.